modules/grizzly/src/main/java/org/glassfish/grizzly/nio/NIOConnection.java

```diff
@@ -865,9 +864,7 @@ protected static void notifyProbesError(NIOConnection connection,
     private void notifyCloseListeners(final CloseReason closeReason) {
         final org.glassfish.grizzly.CloseType closeType =
                 closeReason.getType();
-        
-        org.glassfish.grizzly.CloseListener closeListener;
-        while ((closeListener = closeListeners.poll()) != null) {
+        for(org.glassfish.grizzly.CloseListener closeListener : closeListeners){
```

Comment:
Similar comment here as I made for the 2.3.x changes.  The data structure should probably be cleared out.